### PR TITLE
Add support for nextjs ssr on ui express server, move package page to react

### PR DESCRIPTION
**Pull request by wernst**

### CODE REVIEW

**Comment by wernst** [comment]
Going to fix conflicts, didnt rebase before pushing

File: server/ui.js [hunk 1]
```diff
@@ -204,6 +204,13 @@ function packageRoutes(app) {
     app.post('/dependencies/add', security.verifyAuthToken, bodyParser.json(), async function (req, res) {
         const dependencies = req.body["pack-names"];
         const isDev = req.body["dev-check"] || false;
+
+        if(!dependencies) {
```
Comment (matlin) [inline server/ui.js:208]: `400` (i.e. "Bad Request") might be better

**Review by matlin — CHANGES_REQUESTED** [review]
This is amazing dude. Going to be 100X easier to build the frontend

**Comment by matlin** [comment]
I'm gonna merge but we can fix status codes in another commit.

**Pull request merged**